### PR TITLE
chore: get rid of circleci and node v20 upgrade

### PR DESCRIPTION
**Pull request by kptdobe**

### CODE REVIEW

File: .github/workflows/semantic-release.yaml [hunk 1]
```diff
@@ -7,16 +7,17 @@ on:
 jobs:
   build:
     runs-on: ubuntu-latest
-    if: "!contains(github.event.head_commit.message, '[skip ci]')"
+    if: ${{ !contains(github.event.head_commit.message, '[skip ci]') }}
```
Comment (tripodsan) [inline .github/workflows/semantic-release.yaml:10]: this is no longer required. since 2021, this is automatically respected

Comment (kptdobe) [reply .github/workflows/semantic-release.yaml:10]: good to learn, we have this all over the place :)

**Comment by kptdobe** [comment]
> we have new a combined file for build and release: https://github.com/adobe/franklin-library/blob/4bd14ceb9fb48e32b35bc5a203aaf69b03a61a58/.github/workflows/main.yaml

Humm... what if I prefer having 2 separate files ? This is so much easier to read, especially when each script grow... (not the case here, I agree).

**Review by tripodsan — APPROVED** [review]

**Comment by kptdobe** [comment]
Ok, I have merged the 2 workflows in a single file. I have tried to optimise the code but you cannot get rid of the `actions/checkout@v4`, `actions/setup-node@v4` and `npm ci` steps (easily).

**Pull request merged**